### PR TITLE
feat(http/unstable): `parallel-file-server` module

### LINKED ISSUE #5681: to-do: make `@std/http/file-server` compatible with `deno serve --parallel`
`deno serve --parallel` was implemented in https://github.com/denoland/deno/pull/24920. `@std/http/file-server` should be made to use this sub-command and take advantage of `--parallel`. Benchmarks would also be great.

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #5681

### CODE REVIEW

File: http/parallel_file_server.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * ```sh
+ * deno serve --allow-read --parallel --port=8080 --hostname=1.2.3.4 jsr:@std/http/parallel-file-server
+ * ```
+ *
+ * @example Enable TLS
+ *
+ * By setting the `--cert` and `--key` flags.
+ *
+ * ```sh
+ * deno serve --allow-read --parallel --cert=cert.pem --key=key.pem jsr:@std/http/parallel-file-server
```
Comment (kt3k) [inline http/parallel_file_server.ts:33]: This command seems to show the error:

```
$ deno serve --allow-read --parallel --cert http/testdata/tls/localhost.crt --key http/testdata/tls/localhost.key http/file_server.ts http/parallel_file_server.ts 
error: unexpected argument '--key' found

  tip: to pass '--key' as a value, use '-- --key'

Usage: deno serve --parallel --cert <FILE> [SCRIPT_ARG]...
```

File: http/parallel_file_server.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * ```
+ *
+ * @example Disable directory listing
+ *
+ * By setting the `--no-dir-listing` flag.
+ *
+ * ```sh
+ * deno serve --allow-read --parallel --no-dir-listing jsr:@std/http/parallel-file-server
```
Comment (kt3k) [inline http/parallel_file_server.ts:41]: `--no-dir-listing` needs to be after the script name
```suggestion
 * deno serve --allow-read --parallel jsr:@std/http/parallel-file-server --no-dir-listing
```

File: http/parallel_file_server.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * ```
+ *
+ * @example Disable showing dotfiles
+ *
+ * By setting the `--no-dotfiles` flag.
+ *
+ * ```sh
+ * deno serve --allow-read --parallel --no-dotfiles jsr:@std/http/parallel-file-server
```
Comment (kt3k) [inline http/parallel_file_server.ts:49]:
```suggestion
 * deno serve --allow-read --parallel jsr:@std/http/parallel-file-server --no-dotfiles
```

File: http/parallel_file_server.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+ * ```
+ *
+ * @example Enable CORS
+ *
+ * By setting the `--cors` flag.
+ *
+ * ```sh
+ * deno serve --allow-read --parallel --cors jsr:@std/http/parallel-file-server
```
Comment (kt3k) [inline http/parallel_file_server.ts:57]:
```suggestion
 * deno serve --allow-read --parallel jsr:@std/http/parallel-file-server --cors
```

File: http/parallel_file_server.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+ * ```
+ *
+ * @example Enable verbose mode
+ *
+ * By setting the `--verbose` flag.
+ *
+ * ```sh
+ * deno serve --allow-read --parallel --verbose jsr:@std/http/parallel-file-server
```
Comment (kt3k) [inline http/parallel_file_server.ts:65]:
```suggestion
 * deno serve --allow-read --parallel jsr:@std/http/parallel-file-server --verbose
```

File: http/parallel_file_server.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ * ```
+ *
+ * @example Set custom headers
+ *
+ * By setting the `--header` flag.
+ *
+ * ```sh
+ * deno serve --allow-read --parallel --header=Content-Type:text/plain jsr:@std/http/parallel-file-server
```
Comment (kt3k) [inline http/parallel_file_server.ts:73]:
```suggestion
 * deno serve --allow-read --parallel jsr:@std/http/parallel-file-server --header=Content-Type:text/plain
```

**Comment by kt3k** [comment]
Should we add some warning about that the script is still unstable?

**Comment by iuioiua** [comment]
There are multiple ways we can go about having a HTTP file server work in parallel. I'm not sure this approach is best as I'm not sure having a whole new endpoint is the best way to go. Alternatively, we could have `--parallel` just work with `Deno.serve()`, which would make this work a lot simpler.

**Pull request closed without merging**